setup.py

```diff
@@ -89,29 +90,29 @@ def generate_build_ext(pkg_dir: str, pkg_name: str):
         ("EIGEN_MPL2_ONLY", "1"),  # only MPL-2 part of eigen3
         ("POWER_GRID_MODEL_USE_MKL_AT_RUNTIME", 1),  # use mkl runtime loading
     ]
+    pkg_bin_dir = pkg_dir / "src" / pkg_name
 
     # remove old extension build
-    shutil.rmtree(os.path.join(pkg_dir, "build"), ignore_errors=True)
+    shutil.rmtree(pkg_dir / "build", ignore_errors=True)
```

Comment:
Do we want to ignore errors? If we expect errors in some cases, it might be good to write some comments about the expected errors.

Reply:
Suggestions:
```python
build_dir = pkg_dir / "build"
if build_dir.exists():
    shutil.rmtree(build_dir)
```